src/function/scalar/string/like.cpp

```diff
@@ -1,13 +1,70 @@
 #include "duckdb/common/exception.hpp"
 #include "duckdb/common/vector_operations/vector_operations.hpp"
 #include "duckdb/function/scalar/string_functions.hpp"
+#include "utf8proc.hpp"
 
 using namespace std;
 
 namespace duckdb {
 
 static bool like_operator(const char *s, const char *pattern, const char *escape);
 
+template <bool IS_UPPER>
+static std::unique_ptr<string_t> to_upper_case(Vector &result, const char *input_data, idx_t input_length) {
+	// first figure out the output length
+	// optimization: if only ascii then input_length = output_length
+	idx_t output_length = 0;
+	for (idx_t i = 0; i < input_length;) {
+		if (input_data[i] & 0x80) {
+			// unicode
+			int sz = 0;
+			int codepoint = utf8proc_codepoint(input_data + i, sz);
+			int converted_codepoint = IS_UPPER ? utf8proc_toupper(codepoint) : utf8proc_tolower(codepoint);
+			int new_sz = utf8proc_codepoint_length(converted_codepoint);
+			assert(new_sz >= 0);
+			output_length += new_sz;
+			i += sz;
+		} else {
+			// ascii
+			output_length++;
+			i++;
+		}
+	}
+	std::unique_ptr<string_t> result_str = StringVector::EmptyStringPtr(result, output_length);
+	auto result_data = result_str->GetData();
+
+	for (idx_t i = 0; i < input_length;) {
+		if (input_data[i] & 0x80) {
+			// non-ascii character
+			int sz = 0, new_sz = 0;
+			int codepoint = utf8proc_codepoint(input_data + i, sz);
+			int converted_codepoint = IS_UPPER ? utf8proc_toupper(codepoint) : utf8proc_tolower(codepoint);
+			const auto success = utf8proc_codepoint_to_utf8(converted_codepoint, new_sz, result_data);
+			assert(success);
+			result_data += new_sz;
+			i += sz;
+		} else {
+			// ascii
+			*result_data = IS_UPPER ? toupper(input_data[i]) : tolower(input_data[i]);
+			result_data++;
+			i++;
+		}
+	}
+	result_str->Finalize();
+	return result_str;
+}
+
+struct PrepILike {
+	template<class TA>
+	static inline std::unique_ptr<TA> Operation(Vector& pattern) {
```

Comment:
I can see why this prepping function is useful, but I think it is not generic enough to warrant adding to the BinaryExecutor/TernaryExecutor/etc. If you want to optimize for constant patterns I would rather you look at the way this is done for regular expressions (see `regexp.cpp`, regexp_matches_function, the `bind` function is used to do pre-processing on a constant regexp).

You also need to be careful here as I believe this code is not correct: the pattern is not required to be a constant. What happens if you do:

```sql
SELECT * FROM tbl WHERE i LIKE j;
```

Or even:
```sql
SELECT * FROM tbl WHERE 'hello' LIKE j;
```

Reply:
JFYI: all will be is OK. `ExecuteFlat` worked normally

Reply:
y're right, it's not working if j is column's table. Thanks for info